browser/resources/settings/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts

```diff
@@ -1,151 +1,163 @@
 
-    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this);
-    this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').addEventListener(
-      'click', this.saveOnExitSettingsCallback_);
+    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this)
+    this.shadowRoot!.querySelector('#saveOnExitSettingsConfirm')!
+      .addEventListener('click', this.saveOnExitSettingsCallback_)
 
-    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this);
+    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this)
   }
 
   override disconnectedCallback() {
     super.disconnectedCallback()
 
     if (this.onSelectedPageChangedCallback_) {
       this.$.pages.removeEventListener('selected-item-changed',
-        this.onSelectedPageChangedCallback_);
-      this.onSelectedPageChangedCallback_ = null;
+        this.onSelectedPageChangedCallback_)
+      this.onSelectedPageChangedCallback_ = null
     }
 
-    if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#on-exit-tab').removeEventListener(
-        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_);
-      this.updateSaveButtonStateCallback_ = null;
+    if (this.updateSaveButtonStateCallback_) {
+      this.shadowRoot!.querySelector('#on-exit-tab')!.removeEventListener(
+        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_)
+      this.updateSaveButtonStateCallback_ = null
     }
 
     if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').removeEventListener(
-        'click', this.saveOnExitSettingsCallback_);
-      this.saveOnExitSettingsCallback_ = null;
+      this.shadowRoot!.
+        querySelector('#saveOnExitSettingsConfirm')!.
+        removeEventListener('click', this.saveOnExitSettingsCallback_)
+      this.saveOnExitSettingsCallback_ = null
     }
 
-    this.onClearBraveAdsDataClickHandler_ = () => {};
+    this.onClearBraveAdsDataClickHandler_ = () => {}
   }
 
 /**
   * Updates the text of a browsing data counter corresponding to the given
   * preference.
   * @param {string} prefName Browsing data type deletion preference.
   * @param {string} text The text with which to update the counter
-  * @private
   */
-  updateOnExitCountersText(prefName, text) {
+  private updateOnExitCountersText(prefName: string, text: string) {
     // Data type deletion preferences are named "browser.clear_data.<datatype>".
     // Strip the common prefix, i.e. use only "<datatype>".
-    const matches = prefName.match(/^browser\.clear_data\.(\w+)$/);
-    this.shadowRoot.querySelector('#on-exit-tab').setCounter(matches[1], text);
+    const matches = prefName.match(/^browser\.clear_data\.(\w+)$/)
+    if (matches) {
+      this.shadowRoot!.
+        querySelector<SettingsBraveClearBrowsingDataOnExitPageElement>(
+          '#on-exit-tab')!.setCounter(matches[1], text)
```

Comment:
out of interest, do you know what was happening before when `matches` was falsey?

Reply:
It was telling me this:

```RuntimeError: Command 'C:\work\brave-browser\src\third_party\node\win\node.exe C:\work\brave-browser\src\third_party\node\node_modules\typescript\bin\tsc --project gen/chrome/browser/resources/settings\tsconfig_build_ts.json' failed
gen/chrome/browser/resources/settings/preprocessed/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts:112:37 - error TS18047: 'matches' is possibly 'null'.

112         '#on-exit-tab')!.setCounter(matches[1], text)
                                        ~~~~~~~


Found 1 error in gen/chrome/browser/resources/settings/preprocessed/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts:112
```

Reply:
What you've changed it to is definitely correct! I was just wondering what happened in the old code when it was null :D 

I guess it would just crash?